service/src/java/org/apache/hive/service/cli/operation/Operation.java

```diff
@@ -367,13 +367,13 @@ protected void validateFetchOrientation(FetchOrientation orientation,
       EnumSet<FetchOrientation> supportedOrientations) throws HiveSQLException {
     if (!supportedOrientations.contains(orientation)) {
       throw new HiveSQLException("The fetch type " + orientation.toString() +
-          " is not supported for this resultset", "HY106");
+          " is not supported for this resultset", "HY106", queryState.getQueryId());
     }
   }
 
-  protected HiveSQLException toSQLException(String prefix, CommandProcessorException e) {
+  protected HiveSQLException toSQLException(String prefix, CommandProcessorException e, QueryState queryState) {
```

Comment:
Why are you having ``queryState`` as a parameter & then extracting ``queryId`` from it, we can directly have ``queryId`` as a parameter? as in case of constructor of ``HiveSQLException``